cmd/bisync/bisync_test.go

```diff
@@ -0,0 +1,1249 @@
+	`^DEBUG : .*$`, dropMe,
+	// ignore dropbox info messages
+	`^NOTICE: too_many_(requests|write_operations)/\.*: Too many requests or write operations.*$`, dropMe,
+	`^NOTICE: Dropbox root .*?: Forced to upload files to set modification times on this backend.$`, dropMe,
+	`^INFO  : .*?: src and dst identical but can't set mod time without deleting and re-uploading$`, dropMe,
+}
+
+// Some dry-run messages differ depending on the particular remote.
+var dryrunReplacements = []string{
+	`^(NOTICE: file5.txt: Skipped) (copy|update modification time) (as --dry-run is set [(]size \d+[)])$`,
+	`$1 copy (or update modification time) $3`,
+}
+
+// Some groups of log lines may appear unordered because rclone applies
+// many operations in parallel to boost performance.
+var logHoppers = []string{
+	// Test case `dry-run` produced log mismatches due to non-deterministic
+	// order of captured dry-run info messages.
+	`NOTICE: \S+?: Skipped (?:copy|move|delete|copy \(or [^)]+\)|update modification time) as --dry-run is set \(size \d+\)`,
+
+	// Test case `extended-filenames` detected difference in order of files
+	// with extended unicode names between Windows and Unix or GDrive,
+	// but the order is in fact not important for success.
+	`(?:INFO  |NOTICE): - Path[12] +File (?:was deleted|is new|is newer|is OLDER) +- .*`,
+
+	// Test case `check-access-filters` detected listing miscompares due
+	// to indeterminate order of rclone operations in presence of multiple
+	// subdirectories. The order inconsistency initially showed up in the
+	// listings and triggered reordering of log messages, but the actual
+	// files will in fact match.
+	`ERROR : - +Access test failed: Path[12] file not found in Path[12] - .*`,
+
+	// Test case `resync` suffered from the order of queued copies.
+	`(?:INFO  |NOTICE): - Path2    Resync will copy to Path1 +- .*`,
+}
+
+// Some log lines can contain Windows path separator that must be
+// converted to "/" in every matching token to match golden logs.
+var logLinesWithSlash = []string{
+	`\(\d\d\)  : (touch-glob|touch-copy|copy-file|copy-as|copy-dir|delete-file) `,
+	`INFO  : - Path[12] +Queue copy to Path[12] `,
+	`INFO  : Synching Path1 .*? with Path2 `,
+	`INFO  : Validating listings for `,
+}
+var regexFixSlash = regexp.MustCompile("^(" + strings.Join(logLinesWithSlash, "|") + ")")
+
+// Command line flags for bisync test
+var (
+	argTestCase  = flag.String("case", "", "Bisync test case to run")
+	argRemote2   = flag.String("remote2", "", "Path2 for bisync tests")
+	argNoCompare = flag.Bool("no-compare", false, "Do not compare test results with golden")
+	argNoCleanup = flag.Bool("no-cleanup", false, "Keep test files")
+	argGolden    = flag.Bool("golden", false, "Store results as golden")
+	argDebug     = flag.Bool("debug", false, "Print debug messages")
+	argStopAt    = flag.Int("stop-at", 0, "Stop after given test step")
+	// Flag -refresh-times helps with Dropbox tests failing with message
```

Comment:
@ncw 

Dropbox needs `RefreshTimes` for tests to pass. Probably other backends will need more specific flags.
I don't know how to solve it cleanly.